### ISSUE TITLE
To-Do Items

### ISSUE THREAD

**Issue by GageDeZoort** [GageDeZoort/gnn_tracking#42]
- [ ] Add focal loss and relevant hyperparameters.
- [ ] Remove `predict_track_params` flag from trainer and models.
- [ ] Fix the point cloud TCN model.
- [ ] Add separate encoders / GNN layers for edge classification and object condensation in the graph TCN model.
- [ ] Add loss function weights (fixed by default, but we could add them as learnable parameters later) to the trainer. 
- [ ] Add trainer plots to the plotter class. 
- [ ] Separate the attractive and repulsive potentials if possible. 
- [ ] Fix the numerator and/or denominator in the graph building efficiency measurements; they seem to max out at 99% right now.

**Comment by klieret** [comment]
I've made these into separate issues so we can assign ourselves / keep easier track with what PRs address what etc.
